theme/material/vaadin-text-field-styles.js

```diff
@@ -0,0 +1,221 @@
+      color: inherit;
+      background-color: transparent;
+      /* Disable default invalid style in Firefox */
+      box-shadow: none;
+    }
+
+    /* TODO: the text opacity should be 42%, but the disabled style is 38%.
+      Would need to introduce another property for it if we want to be 100% accurate. */
+    [part='value']::placeholder {
+      color: var(--material-disabled-text-color);
+      transition: opacity 0.175s 0.1s;
+      opacity: 1;
+    }
+
+    :host([has-label]:not([focused]):not([invalid]):not([theme='always-float-label'])) [part='value']::placeholder {
+      opacity: 0;
+      transition-delay: 0;
+    }
+
+    /* According to material theme guidelines, helper text should be hidden when error message is set and input is invalid */
+    :host([has-helper][invalid][has-error-message]) [part='helper-text'] {
+      display: none;
+    }
+
+    [part='label'] {
+      width: 133%;
+      transition: transform 0.175s, color 0.175s, width 0.175s;
+      transition-timing-function: ease, ease, step-end;
+    }
+
+    :host([has-helper]) [part='helper-text']::before {
+      content: '';
+      display: block;
+      height: 6px;
+    }
+
+    [part='helper-text'],
+    [part='helper-text'] ::slotted(*) {
+      font-size: 0.75rem;
+      line-height: 1;
+      color: var(--material-secondary-text-color);
+    }
+
+    :host(:hover:not([readonly]):not([invalid])) [part='input-field']::before {
+      opacity: var(--_material-text-field-input-line-hover-opacity, 0.87);
+    }
+
+    :host([focused]:not([invalid])) [part='label'] {
+      color: var(--material-primary-text-color);
+    }
+
+    :host([focused]) [part='input-field']::after,
+    :host([invalid]) [part='input-field']::after {
+      opacity: 1;
+      transform: none;
+      transition: transform 0.175s, opacity 0.175s;
+    }
+
+    :host([invalid]) [part='input-field']::after {
+      background-color: var(--material-error-color);
+    }
+
+    :host([input-prevented]) [part='input-field'] {
+      color: var(--material-error-text-color);
+    }
+
+    :host([disabled]) {
+      pointer-events: none;
+    }
+
+    :host([disabled]) [part='input-field'] {
+      color: var(--material-disabled-text-color);
+    }
+
+    :host([disabled]) [part='input-field']::before {
+      background-color: transparent;
+      background-image: linear-gradient(
+        90deg,
+        var(--_material-text-field-input-line-background-color, #000) 0,
+        var(--_material-text-field-input-line-background-color, #000) 2px,
+        transparent 2px
+      );
+      background-size: 4px 1px;
+      background-repeat: repeat-x;
+    }
+
+    /* Only target the visible floating label */
+    :host([has-label]:not([has-value]):not([focused]):not([invalid]):not([theme~='always-float-label']))
+      [part='label'] {
+      width: 100%;
+      /* IE11 doesn’t work with calc inside the translate function, so we need to have a fixed pixel value instead of 50% + 16px */
```

Comment:
Remove IE11 workaround

Reply:
I tried to use 50% + 16px and that caused regressions in visual tests. Removed the comment for now.

Let's revisit this later, as Material theme in general is not up to date.